src/service/pluginService.ts

```diff
@@ -127,210 +112,147 @@ export class PluginService {
-            ignoredExtensions.includes(ext.name)
-          ) {
-            found = true;
-            break;
-          }
-        }
-        if (!found) {
-          deletedList.push(ext);
-        }
-      }
-    }
-    return deletedList;
+    return localExtensions.filter(
+      ext =>
+        ext.name !== "code-settings-sync" &&
+        !remoteExtensions.map(e => e.name).includes(ext.name) &&
+        !ignoredExtensions.includes(ext.name)
+    );
   }
 
   public static CreateExtensionList() {
-    const list: ExtensionInformation[] = [];
-
-    for (const ext of vscode.extensions.all) {
-      if (ext.packageJSON.isBuiltin === true) {
-        continue;
-      }
-
-      const meta = ext.packageJSON.__metadata || {
-        id: ext.packageJSON.uuid,
-        publisherId: ext.id,
-        publisherDisplayName: ext.packageJSON.publisher
-      };
-      const data = new ExtensionMetadata(
-        meta.galleryApiUrl,
-        meta.id,
-        meta.downloadUrl,
-        meta.publisherId,
-        meta.publisherDisplayName,
-        meta.date
-      );
-      const info = new ExtensionInformation();
-      info.metadata = data;
-      info.name = ext.packageJSON.name;
-      info.publisher = ext.packageJSON.publisher;
-      info.version = ext.packageJSON.version;
-      list.push(info);
-    }
-
-    return list;
+    return vscode.extensions.all
+      .filter(ext => !ext.packageJSON.isBuiltin)
+      .map(ext => {
+        const meta = ext.packageJSON.__metadata || {
+          id: ext.packageJSON.uuid,
+          publisherId: ext.id,
+          publisherDisplayName: ext.packageJSON.publisher
+        };
+        const data = new ExtensionMetadata(
+          meta.galleryApiUrl,
+          meta.id,
+          meta.downloadUrl,
+          meta.publisherId,
+          meta.publisherDisplayName,
+          meta.date
+        );
+        const info = new ExtensionInformation();
+        info.metadata = data;
+        info.name = ext.packageJSON.name;
+        info.publisher = ext.packageJSON.publisher;
+        info.version = ext.packageJSON.version;
+        return info;
+      });
   }
 
   public static async DeleteExtension(
-    item: ExtensionInformation,
-    ExtensionFolder: string
+    extension: ExtensionInformation
   ): Promise<boolean> {
-    const destination = path.join(
-      ExtensionFolder,
-      item.publisher + "." + item.name + "-" + item.version
-    );
-
-    try {
-      await fs.remove(destination);
-      return true;
-    } catch (err) {
-      console.log("Sync : " + "Error in uninstalling Extension.");
-      console.log(err);
-      throw err;
-    }
+    return new Promise<boolean>((resolve, reject) => {
+      try {
+        vscode.commands
+          .executeCommand(
+            "workbench.extensions.uninstallExtension",
+            `${extension.publisher}.${extension.name}`
+          )
+          .then(resolve);
+      } catch (err) {
+        reject(err);
+        throw new Error(err);

```

Comment:
Will this line of code execute when it is rejected? What do you suggest?

Reply:
Yes, once `reject()` is called, the detailed error will be thrown into the console.